### DISCUSSION TITLE
GTD+ addon - dark text color in task panel

### DISCUSSION THREAD

**Discussion by joemcbrooks** [freeplane/freeplane#1589] [Help]
I like the GTD+ addon but I am struggling with the text color of items in the sidebar (tool panel) when I'm using a dark theme like Flat Darcula. Unlike the other tabs like format and attributes where the text color has adapted to the dark background the text color in the GTD panel remains black on dark background. I tried to change some text related CSS values in freeplaneGTD.groovy - but with no success so far. I guess the setting is done elsewhere? 
![image](https://github.com/freeplane/freeplane/assets/6098355/1dfd00dd-6539-475b-9c7e-df8e31a6f670)

**Comment by zpuric** [comment]
Hi @joemcbrooks,

I found a fix for this on my side.

The problem seems to come from the hardcoded `Color.BLACK` here in `GtdReportViewController.groovy`:

```groovy
JLabel taskLabel = label(text: actionText, foreground: overdue ? Color.RED : Color.BLACK)
```
I replaced it with:
```groovy
Color normalTextColor = UIManager.getColor("Label.foreground") ?: Color.WHITE 
JLabel taskLabel = label(text: actionText, foreground: overdue ? Color.RED : normalTextColor)
```
This made the GTD Next Actions sidebar text display correctly for me in both the light and dark Freeplane UI themes.

If the modified script does not seem to be picked up immediately, it may help to delete the `compiledscripts` and `compiledscripts2 ` cache folders and then restart Freeplane.

Before:
<img width="529" height="395" alt="image" src="https://github.com/user-attachments/assets/8d821304-be5a-4814-aa65-f4e6c73a5e8c" />
After:
<img width="535" height="396" alt="image" src="https://github.com/user-attachments/assets/8eacf329-2a33-4880-a912-23a85a5b4e11" />

I’m also attaching my modified `GtdReportViewController.groovy` file in case it is useful for testing or comparison.
[GtdReportViewController.zip](https://github.com/user-attachments/files/26189357/GtdReportViewController.zip)
